types/index.d.ts

```diff
@@ -121,7 +121,8 @@ declare module 'mongoose' {
    * Returns true if Mongoose can cast the given value to an ObjectId, or
    * false otherwise.
    */
-  export function isValidObjectId(v: any): boolean;
+  export function isValidObjectId(v: Types.ObjectId): true;
+  export function isValidObjectId(v: unknown): boolean;
```

Comment:
changed back to unknown.

two definitions, because if you specifiy that v is of Types.ObjectId always returns true, than typescript can determine, that an if clause is always truthy and you could technically even see if there is a dead branch because the else branch is never "hit".